apps/root/src/pages/aggregator/swap-container/components/swap-settings/index.tsx

```diff
@@ -107,146 +184,179 @@ const SwapSettings = ({ shouldShow, onClose }: SwapSettingsProps) => {
     trackEvent('Aggregator - Set default settings');
   };
 
-  const handleToggleDex = (checked: boolean, id: string) => {
-    const newDisabledDexes = [...disabledDexes];
-
-    if (newDisabledDexes.includes(id)) {
-      newDisabledDexes.splice(newDisabledDexes.indexOf(id), 1);
-      trackEvent('Aggregator - Disable dex', { source: id });
-    } else {
-      newDisabledDexes.push(id);
-      trackEvent('Aggregator - Enable dex', { source: id });
-    }
+  const dexOptions: OptionsMenuOption[] = Object.keys(dexes).map((dexKey) => ({
+    label: '',
+    type: OptionsMenuOptionType.option,
+    closeOnClick: false,
+    icon: (
+      <FormGroup key={dexKey}>
+        <FormControlLabel
+          control={<Checkbox onChange={() => handleToggleDex(dexKey)} checked={!disabledDexes.includes(dexKey)} />}
+          label={dexes[dexKey].name}
+        />
+      </FormGroup>
+    ),
+  }));
 
-    dispatch(setDisabledDexes(newDisabledDexes));
-  };
+  const sortQuotesOptions: OptionsMenuOption[] = Object.entries(SWAP_ROUTES_SORT_OPTIONS).map(
+    ([optionKey, optionData]) => ({
+      label: intl.formatMessage(optionData.label),
+      type: OptionsMenuOptionType.option,
+      control: (
+        <Typography variant="bodySmall">
+          <Tooltip title={intl.formatMessage(optionData.help)} arrow placement="top">
+            <HelpOutlineIcon fontSize="small" />
+          </Tooltip>
+        </Typography>
+      ),
+      onClick: () => setQuoteSorting(optionKey as SwapSortOptions),
+    })
+  );
 
   return (
-    <Slide direction="up" in={shouldShow} mountOnEnter unmountOnExit>
+    <Slide
+      direction="up"
+      in={shouldShow}
+      mountOnEnter
+      unmountOnExit
+      onExited={() => setShouldShowFirstStep(true)}
```

Comment:
why is this needed?

Reply:
This is the same approach I did on transaction steps. The height for swap-settings is different from the height of swap first step. So having the <Slide /> as absolute with overflow: hidden in the container would not work anymore. With this implementation, the First step unmounts when the settings Slide started entering, and mounts when the Slide has exited